pkg/apis/v1beta1/nodepool.go

```diff
@@ -35,18 +37,18 @@ type NodePoolSpec struct {
 	// Deprovisioning contains the parameters that relate to Karpenter's deprovisioning logic
 	// +kubebuilder:default={"consolidationTTL": "15s","consolidationPolicy": "WhenUnderutilized", "expirationTTL": "90d"}
 	// +optional
-	Deprovisioning Deprovisioning `json:"deprovisioning"`
+	Deprovisioning Deprovisioning `json:"deprovisioning" hash:"ignore"`
```

Comment:
Do we want to hash the full spec, or just hash the template, so you don't need all of these fields.